src/scripts/main.js

```diff
@@ -1,3 +1,300 @@
+
+  addRandomNumber(getRandomNumber());
+  addRandomNumber(getRandomNumber());
+};
+
+function getAllCells() {
+  return [...document.querySelectorAll('.field-cell')];
+};
+
+function getFreeCells() {
+  return [...document.querySelectorAll('.field-cell')]
+    .filter(cell => cell.innerText === '');
+};
+
+function moveCells(firsCell, secondCell) {
+  const firstValue = Number(firsCell.innerText);
+  const secondValue = Number(secondCell.innerText);
+
+  if (firstValue !== 0 && secondValue === 0) {
+    firsCell.className = 'field-cell';
+    firsCell.innerText = '';
+    secondCell.innerText = firstValue;
+    secondCell.classList.add(`field-cell--${firstValue}`);
+    actionCounter++;
+  };
+
+  if (
+    firstValue !== 0 && firstValue === secondValue
+    && !blockedCells.includes(firsCell)
+  ) {
+    firsCell.className = 'field-cell';
+    firsCell.innerText = '';
+    secondCell.innerText = firstValue * 2;
+    secondCell.classList.add(`field-cell--${firstValue * 2}`);
+    blockedCells.push(secondCell);
+
+    scoreCounter += Number(secondCell.innerText);
+    scoreInfo.innerText = scoreCounter;
+    actionCounter++;
+
+    if (secondCell.innerText === '2048') {
+      winGame();
+    };
+  };
+};
+
+const moveRight = () => {
+  [...gameField.rows].forEach(row => {
+    const cells = [...row.cells];
+
+    for (let i = 2; i >= 0; i--) {
+      const firsCell = cells[i];
+      const secondCell = cells[i + 1];
+
+      moveCells(firsCell, secondCell);
+    };
+  });
+};
+
+const moveLeft = () => {
+  [...gameField.rows].forEach(row => {
+    const cells = [...row.cells];
+
+    for (let i = 1; i < 4; i++) {
+      const firsCell = cells[i];
+      const secondCell = cells[i - 1];
+
+      moveCells(firsCell, secondCell);
+    };
+  });
+};
+
+const moveUp = () => {
+  const columns = [];
+  const fieldRows = [...gameField.rows];
+
+  for (let i = 0; i < fieldRows.length; i++) {
+    const column = [];
+
+    fieldRows.forEach(row => {
+      column.push([...row.cells][i]);
+    });
+
+    columns.push(column);
+  };
+
+  columns.forEach(column => {
+    for (let i = 1; i < 4; i++) {
+      const firsCell = column[i];
+      const secondCell = column[i - 1];
+
+      moveCells(firsCell, secondCell);
+    };
+  });
+};
+
+const moveDown = () => {
+  const columns = [];
+  const fieldRows = [...gameField.rows];
+
+  for (let i = 0; i < fieldRows.length; i++) {
+    const column = [];
+
+    fieldRows.forEach(row => {
+      column.push([...row.cells][i]);
+    });
+
+    columns.push(column);
+  };
+
+  columns.forEach(column => {
+    for (let i = 2; i >= 0; i--) {
+      const firstCell = column[i];
+      const secondCell = column[i + 1];
+
+      moveCells(firstCell, secondCell);
+    };
+  });
+};
+
+startButton.addEventListener('click', (e) => {
+  startGame();
+});
+
+document.addEventListener('keydown', (e) => {
+  const freeCells = getFreeCells();
+
+  switch (e.key) {
+    case 'ArrowRight':
+      moveRight();
+      moveRight();
```

Comment:
Repeated calls to the moveRight() function can be replaced with a loop or a function that accepts a parameter for the number of times to move.
Apply for similar cases